foxpuppet/windows.py

```diff
@@ -2,44 +2,50 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from contextlib import contextmanager
-
-from selenium.webdriver.common.by import By
 from selenium.webdriver.support.wait import WebDriverWait
 
 
-class Windows(object):
+class WindowManager(object):
 
     def __init__(self, selenium):
         self.selenium = selenium
+        self.windows = None
+
+        self.populate_windows()
+
+    def populate_windows(self):
+        self.windows = []
+        for handle in self.selenium.window_handles:
+            self.windows.append(BaseWindow(self.selenium, handle))
```

Comment:
This method in Firefox Puppeteer doesn't create BaseWindow instances only. It checks which type of window it is and automatically creates the correct class instance.